src/components/builder/values/ValuesTable.tsx

```diff
@@ -0,0 +1,109 @@
+import {Option} from '@open-formulieren/types/lib/formio/common';
+import {FieldArray, useFormikContext} from 'formik';
+import {FormattedMessage} from 'react-intl';
+
+import {Component} from '@/components/formio';
+import {PathsForValueType} from '@/types';
+
+import OptionRow from './OptionRow';
+
+export interface ValuesTableProps<S> {
+  /**
+   * Name of the field holding the (component) values, e.g. `values` or `data.values`.
+   */
+  name: PathsForValueType<S, Option[]> & string;
+}
+
+/**
+ * Manage a set of option values/labels for a component.
+ *
+ * Formio typically allows you to specify manual options/values for preset options,
+ * like the dropdown for a select, the values + labels for a radio choice or multiple
+ * checkboxes.
+ *
+ * This component is a generic component, you must use it by passing the relevant
+ * concrete schema for `name` prop autocomplete and type checking, e.g.:
+ *
+ *
+ *     <ValuesTable<SelectComponentSchema> name="data.values" />
+ *
+ * Translations for every option label are (to be) managed in a dedicated translations
+ * component.
+ */
+function ValuesTable<S>({name}: ValuesTableProps<S>) {
+  const {getFieldProps} = useFormikContext();
+  const {value: options = []} = getFieldProps<Option[] | undefined>(name);
+
+  return (
+    <Component type="datagrid">
+      <FieldArray name={name}>
+        {arrayHelpers => (
+          <table className="table table-bordered">
+            <thead>
+              <tr>
+                <th />
+
+                <th>
+                  <FormattedMessage
+                    description="Option label table header/label"
+                    defaultMessage="Label"
+                  />
+                </th>
+
+                <th className="field-required">
+                  <FormattedMessage
+                    description="Option value table header/label"
+                    defaultMessage="Value"
+                  />
+                </th>
+
+                <th>
+                  <span className="sr-only">
+                    <FormattedMessage
+                      description="Option add/remove table header/label"
+                      defaultMessage="Add/remove"
+                    />
+                  </span>
+                </th>
+              </tr>
+            </thead>
+
+            <tbody>
+              {options.map((_, index) => (
+                <tr key={index}>
+                  <OptionRow name={name} index={index} arrayHelpers={arrayHelpers} />
```

Comment:
so we're interested in both `name` and `index` for a single option:

1. `name` combined with `index` determines the values namespace for each option, leading to `${name}[${index}].label` and `${name}[${index}].value`
2. the `index` is used in combination with `FormikValues[$name]`, which contains the entire list of all the options for the component - to check whether an option is the last or first element, as that possibly disables one of the sort controls
3. `name` is the only part that is dynamic about this, in practice it's `"data.values" | "values"`, the rest is fixed because of the `Option[]` type.

alternatively we'd need to calculate `isFirst` and `isLast` in the parent component and pass that down to the `OptionRow`, and then the option row prop types get a bit weird since if you have > 2 options, isLast/isFirst cannot both be true at the same time, I picked this approach because it allows you to pass the atomic data and calculate any derivations in the component itself.